### DISCUSSION TITLE
Is there a way to force Microstream to load Lazy References?

### DISCUSSION THREAD

**Discussion by Sandared** [microstream-one/microstream#261] [Q&A] [answered]
Hi,

I already opened a support Ticket regarding this topic, but I figured this might be of interest to others as well, therefore I'm crossposting here too:

Currently I'm trying to test our data schema which we persist with Microstream. For this I usually write a test where I create an Object in memory, then restart the DB, then fetch the same object again and then recursively compare those two via AssertJ's recursive comparison.

Now we have made some fields of such an object `Lazy`, which breaks such tests, as the fields that are marked as Lazy are not loaded directly after we restart the DB.

Therefore my question is: 
Is there a way to force Microstream to load all Lazy References on Startup, e.g., by using a special configuration?
Or is there another, better way to deal with such a scenario that I'm not aware of?

Kind regards,
Thomas

**Comment by hg-ms** [comment, marked as answer]
Hello,

the LazyReferenceManager has an iterate method that iterates over all registered LazyReferences.
you could use this method to load all lazy references.

Optionally you can also deactivate the LazyReferenceManager at any time after starting the storage to prevent it from clearing LazyReferences automatically if necessary.
```
final EmbeddedStorageManager storage = EmbeddedStorage.start();
LazyReferenceManager.get().iterate( lazyRef -> lazyRef.get());
LazyReferenceManager.get().stop();
```

Best regards

**Reply by Sandared** [reply]
Thanks for that quick answer :) 👍 
Did solve my problem!
